clang/test/AST/HLSL/StructuredBuffers-AST.hlsl

```diff
@@ -89,12 +89,12 @@ RESOURCE<float> Buffer;
 // CHECK-SAME{LITERAL}: [[hlsl::raw_buffer]]
 // CHECK-SAME{LITERAL}: [[hlsl::contained_type(element_type)]]
 
-// CHECK-SUBSCRIPT: CXXMethodDecl {{.*}} operator[] 'const element_type &(unsigned int) const'
+// CHECK-SUBSCRIPT: CXXMethodDecl {{.*}} operator[] 'hlsl_device element_type &const (unsigned int) const'
```

Comment:
I don't think this is correct. The method needs to return a reference to a constant element_type value (for read-only resources), not a constant reference to element_type value. Please correct me if I am wrong, but I believe it must be:

`operator[] 'hlsl_device const element_type &(unsigned int) const`

Reply:
Good catch. I was misreading what the const applied to.